### DISCUSSION TITLE
How to upgrade longhorn having deviation from default settings

### DISCUSSION THREAD

**Discussion by masifpak** [longhorn/longhorn#5371] [Q&A]
## What's the task? Please describe

I am using longhorn 1.2.4 deployed from rancher UI. Now I need to update it from 1.2.4 to 1.3.2. When I read the following section I alert. Actually I reduces no. of replicas from 3 to 2. Now how can I upgrade. It will fail as describe here.

![image](https://user-images.githubusercontent.com/12061167/216044834-ad391171-94a2-4e49-b760-f2f2232df47a.png)

**Comment by weizhe0422** [comment]
Would you mind providing more information on the relevant error message displayed in the console when starting the upgrade?

**Comment by masifeddress** [comment]
Thanks @weizhe0422 Actually I did not start yet. Before upgrade, I was planning. When I read this I plan to get extra safty measures. 
I am asking what to do if I update no. of replicas in my storage class. Because documentation is that it will throw error.

**Comment by weizhe0422** [comment]
IIUC, I think it might can rename this updated storage class, and modifying the annotation `storageclass.kubernetes.io/is-default-class` in the configmap `longhorn-storageclass` data to `false`.

 `kubectl -n longhorn-system edit configmap longhorn-storageclass`

Then deploy a default storage class via this [configMap](https://github.com/longhorn/longhorn/blob/3a30bd8fedaa623af3f061d9f5475fd54b894043/deploy/longhorn.yaml#L43-L70).
